File: LeetCode Solutions/March 2021 LeetCoding Challenges/Keys and Rooms/Keys and Rooms.java
class Solution {
    public boolean canVisitAllRooms(List<List<Integer>> rooms) {
        boolean[] arr = new boolean[rooms.size()];
        arr[0] = true;
        Stack<Integer> stack = new Stack<>();
        stack.push(0);
        int count = 1;
        while (stack.size() > 0){
            for (int k : rooms.get(stack.pop())){
                if (!arr[k]){
                    stack.push(k);
                    arr[k] = true;
                    count++;
                }
            }
        }
        return rooms.size() == count;
    }
}

File: LeetCode Solutions/Graph Theory I/Keys and Rooms.py
class Solution:
    def canVisitAllRooms(self, rooms: List[List[int]]) -> bool:
        visited = set()
        def helper(i, room):
            visited.add(i)
            print(room)
            for key in room:
                if(key not in visited):
                    helper(key, rooms[key])
        helper(0, rooms[0])
        return len(visited) == len(rooms)